modules/lib/lib-project/src/main/resources/lib/xp/project.js

```diff
@@ -9,49 +45,117 @@ exports.create = function (params) {
+ */
 exports.delete = function (params) {
     var bean = __.newBean('com.enonic.xp.lib.project.DeleteProjectHandler');
-    bean.name = required(params, 'name');
+    bean.id = required(params, 'id');
     return __.toNativeObject(bean.execute());
 };
 
+/**
+ * Returns an existing Content Project.
+ *
+ * @param {object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+
+ *
+ * @returns {object} Content Project.
+ */
 exports.get = function (params) {
     var bean = __.newBean('com.enonic.xp.lib.project.GetProjectHandler');
-    bean.name = required(params, 'name');
+    bean.id = required(params, 'id');
     return __.toNativeObject(bean.execute());
 };
 
-exports.list = function (params) {
+/**
+ * Returns all Content Projects.
+ *
+
+ *
+ * @returns {object[]} Array of Content Projects.
+ */
+exports.list = function () {
     var bean = __.newBean('com.enonic.xp.lib.project.ListProjectsHandler');
     return __.toNativeObject(bean.execute());
 };
 
+/**
+ * Adds permissions to an existing Content Project.
+ *
+ * @param {object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {PermissionsDef} params.permissions Project permissions to add.
+
+ *
+ * @returns {object} Content Project.
+ */
 exports.addPermissions = function (params) {
     var bean = __.newBean('com.enonic.xp.lib.project.AddProjectPermissionsHandler');
-    bean.name = required(params, 'name');
+    bean.id = required(params, 'id');
     bean.permissions = __.toScriptValue(params.permissions);
     return __.toNativeObject(bean.execute());
 };
 
+/**
+ * Removes permissions from an existing Content Project.
+ *
+ * @param {object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {PermissionsDef} params.permissions Project permissions to remove.
+
+ *
+ * @returns {object} Content Project.
+ */
 exports.removePermissions = function (params) {
     var bean = __.newBean('com.enonic.xp.lib.project.RemoveProjectPermissionsHandler');
-    bean.name = required(params, 'name');
+    bean.id = required(params, 'id');
     bean.permissions = __.toScriptValue(params.permissions);
     return __.toNativeObject(bean.execute());
 };
 
+/**
+ * Toggles public/private read access for an existing Content Project.
+ * This will modify permissions on all of the content items inside the repository by adding or removing READ access for `system.everyone`.
+ *
+ * @param {object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {ReadAccessDef} params.readAccess Read access.
+
+ *
+ * @returns {object} Content Project.
+ */
 exports.modifyReadAccess = function (params) {
     var bean = __.newBean('com.enonic.xp.lib.project.ModifyProjectReadAccessHandler');
-    bean.name = required(params, 'name');
+    bean.id = required(params, 'id');
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['required' was used before it was defined.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5511138)